src/openscm_runner/adapters/magicc7/magicc7.py

```diff
@@ -5,12 +5,12 @@
 import os
 from subprocess import check_output  # nosec
 
-import pymagicc
 from scmdata import ScmRun, run_append
 
 from ...progress import progress
 from ...settings import config
 from ..base import _Adapter
+from ._magicc_instances import HAS_PYMAGICC, pymagicc
```

Comment:
I like this way the best. It keeps the conditional importing in one location. Although I would refactor into a separate module. That would avoid any possible cyclic imports in future and the order of imports wouldn't be an issue (they aren't currently, but it seems cleaner)

Reply:
As above